src/cargo/ops/cargo_compile.rs

```diff
@@ -408,13 +408,24 @@ fn scrape_build_config(config: &Config,
 fn scrape_target_config(config: &Config, triple: &str)
                         -> CargoResult<ops::TargetConfig> {
 
+    let get_path_string = |key: &str| -> CargoResult<Option<String>> {
+        Ok(
+            if let Some(path) = try!(config.get_path(key)) {
+                Some(try!(path.into_os_string().into_string()
+                          .map_err(|_| internal("unhandled non-utf8 path"))))
+            } else {
+                None
+            }
+        )
+    };
+
     let key = format!("target.{}", triple);
-    let ar = try!(config.get_string(&format!("{}.ar", key)));
-    let linker = try!(config.get_string(&format!("{}.linker", key)));
+    let ar = try!(get_path_string(&format!("{}.ar", key)));
+    let linker = try!(get_path_string(&format!("{}.linker", key)));
 
     let mut ret = ops::TargetConfig {
-        ar: ar.map(|p| p.0),
-        linker: linker.map(|p| p.0),
+        ar: ar,
+        linker: linker,
```

Comment:
Could these be switch to being stored as a `PathBuf` to avoid the convert-to-utf8 requirement?